common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1055,21 +1085,21 @@ def templates(cls):
         Expects a class attribute template_dir_name that defines the directory
         inside the 'templates' resource directory to pull templates from
         """
-        templates = []
+        templates = {}
         dirname = cls.get_template_dir()
         if dirname is not None:
             for pkg in cls.template_packages:
                 if not resource_isdir(pkg, dirname):
                     continue
-                for template_file in resource_listdir(pkg, dirname):
-                    if not template_file.endswith('.yaml'):
-                        log.warning("Skipping unknown template file %s", template_file)
-                        continue
-                    template_content = resource_string(pkg, os.path.join(dirname, template_file))
-                    template = yaml.safe_load(template_content)
-                    template['template_id'] = template_file
-                    templates.append(template)
-        return templates
+                for template in cls._load_templates_in_dir(resource_filename(pkg, dirname)):
```

Comment:
@keithgg Just me nitpicking: do you think we can rework this so that `get_template_dir` becomes `get_template_dirs` and returns a list of possible template directories so this can iterate over it?

Doing it like this would remove the need for code branches here and under `get_template` and make it easier to plug more directories in the future if we need to. What do you think?

Reply:
Happy to do that. I left the existing `pkg_resources` way of loading to minimise my changes and because I couldn't quite figure out why it was using `pkg_resources` over an `os.listdir`.

Reply:
Corrected.